src/components/dashboard/StakingBanner/index.tsx

```diff
@@ -0,0 +1,128 @@
+import { Typography, Card, SvgIcon, Grid, Button, Box, Stack } from '@mui/material'
+import css from './styles.module.css'
+import Kiln from '@/public/images/common/kiln.svg'
+import StakeIllustrationLight from '@/public/images/common/stake-illustration-light.svg'
+import StakeIllustrationDark from '@/public/images/common/stake-illustration-dark.svg'
+import classNames from 'classnames'
+import { useDarkMode } from '@/hooks/useDarkMode'
+import { useRouter } from 'next/router'
+import NextLink from 'next/link'
+import { OVERVIEW_EVENTS, trackEvent } from '@/services/analytics'
+import useLocalStorage from '@/services/local-storage/useLocalStorage'
+import ExternalLink from '@/components/common/ExternalLink'
+import { useSanctionedAddress } from '@/hooks/useSanctionedAddress'
+import useIsStakingFeatureEnabled from '@/features/stake/hooks/useIsSwapFeatureEnabled'
+import { AppRoutes } from '@/config/routes'
+import useBalances from '@/hooks/useBalances'
+import { formatUnits } from 'ethers'
+import { TokenType } from '@safe-global/safe-gateway-typescript-sdk'
+
+const LOCAL_STORAGE_KEY_HIDE_WIDGET = 'hideStakingBanner'
+const LEARN_MORE_LINK = ''
+const MIN_NATIVE_TOKEN_BALANCE = 32
+
+const StakingBanner = () => {
+  const isDarkMode = useDarkMode()
+  const router = useRouter()
+  const { balances } = useBalances()
+
+  const nativeTokenBalance = balances.items.find((balance) => balance.tokenInfo.type === TokenType.NATIVE_TOKEN)
```

Comment:
I guess it doesn't matter here as the balances won't be that huge of an array, but not having this memoized or not in a separate hook that is executed just once means that this line will be constantly recomputed for whatever reason the widget rerenders.